app/questionnaire/questionnaire_schema.py

```diff
@@ -1385,27 +1385,47 @@ def get_item_anchor(self, section_id: str, list_name: str) -> str | None:
                 if item["for_list"] == list_name and item.get("item_anchor_answer_id"):
                     return f"#{str(item['item_anchor_answer_id'])}"
 
+    def _update_placeholder_dependencies(
+        self, placeholder_answer_ids: set, block: ImmutableDict
+    ) -> None:
+        if placeholder_dependencies := self._get_section_ids_for_answer_ids(
+            answer_ids=placeholder_answer_ids
+        ):
+            # Type Ignore: At this point the section id and block id cannot be None
+            section_id = self.get_section_id_for_block_id(block["id"])
+            self._placeholder_transform_section_dependencies_by_block[section_id][  # type: ignore
+                block["id"]
+            ].update(
+                placeholder_dependencies
+            )
+
     def _populate_placeholder_transform_section_dependencies(self) -> None:
         for block in self.get_blocks():
-            transforms = get_mappings_with_key("transform", data=block)
-            placeholder_answer_ids = {
-                item["identifier"]
-                for transform in transforms
-                if transform["transform"] in TRANSFORMS_REQUIRING_ROUTING_PATH
-                for item in transform["arguments"]["items"]
-                if item.get("source") == "answers"
-            }
-            placeholder_dependencies = self._get_section_ids_for_answer_ids(
-                answer_ids=placeholder_answer_ids
-            )
-            if placeholder_dependencies:
-                # Type Ignore: At this point we section id  and block id cannot be None
-                section_id = self.get_section_id_for_block_id(block["id"])
-                self._placeholder_transform_section_dependencies_by_block[section_id][  # type: ignore
+            answer_ids: list = []
```

Comment:
I guess this is wrong cause the block cannot be CS and GCS at the same time, so:
```python
            if block["type"] == "GrandCalculatedSummary":
                answer_ids = self.get_answer_ids_for_grand_calculated_summary_id(
                    block["id"]
                )
            elif block["type"] == "CalculatedSummary":
                answer_ids = get_calculated_summary_answer_ids(block)
```
or even:
```python
            if block["type"] == "GrandCalculatedSummary":
                answer_ids = self.get_answer_ids_for_grand_calculated_summary_id(
                    block["id"]
                )
            elif block["type"] == "CalculatedSummary":
                answer_ids = get_calculated_summary_answer_ids(block)
            else:
               answer_ids = []
```
or something similar..